ethereum/solidity/contracts/DataRootTuple.sol

```diff
@@ -0,0 +1,14 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.0;
+
+/// @notice A tuple of data root with metadata. Each data root is associated
+///  with a Celestia block height.
+/// @dev `availableDataRoot` in
+///  https://github.com/celestiaorg/celestia-specs/blob/master/src/specs/data_structures.md#header
+struct DataRootTuple {
+    // Celestia block height the data root was included in.
+    // Genesis block is height = 1.
+    uint256 height;
+    // Data root.
```

Comment:
```suggestion
    // Data root: The commitment over the data roots for the range of blocks: previous_heigh -> heigh.
```

Reply:
Wait no, the data root in this struct is a single data root.

Reply:
But we don't have access to them here, no ?
What the contract will receive is a commitment over a set of data roots, no ? And the corresponding height.

Reply:
> What the contract will receive is a commitment over a set of data roots, no ?

Yes. But that's not a `DataRootTuple`. It would be the _root_ of several `DataRootTuple`s.

Reply:
Aaaaaaah now I get it. Thanks.
Wouldn't it make sense to include these data roots also when submitting the commitment ? and checking if the commitment is correct ? Or it would be too expensive ?

Reply:
> Wouldn't it make sense to include these data roots also when submitting the commitment ? and checking if the commitment is correct ?

Nope! Under an honest majority assumption, the root will be correct. That's one of the optimizations we've made over the regular Gravity bridge.